include/Match.h

```diff
@@ -18,330 +18,365 @@ using std::list;
-    double relativeTolerance;
-    set<int> frozenParameters;  // Keep track of degenerate parameters
-    map<string, set<int>> frozenMaps; // Which atoms have which params frozen
-  public:
-    CoordAlign(PixelMapCollection& pmc_,
-	       MCat& mlist_): mlist(mlist_),
-			      pmc(pmc_), 
-			      relativeTolerance(0.001)  {}
-
-    // Re-map Detections using current params - either all or just those being fit.
-    void remap(bool doAll=true); 
-    // Fitting routine: returns chisq of current fit, updates params. and remaps.
-    double fitOnce(bool reportToCerr=true,
-		   bool inPlace=false);	  // Set inPlace to save space, but can't debug singularities
-    // Conduct one round of sigma-clipping.  If doReserved=true, 
-    // then only clip reserved Matches.  If =false, then
-    // only clip non-reserved Matches.
-    int sigmaClip(double sigThresh, bool doReserved=false,
-		  bool clipEntireMatch=false,
-		  bool logging=true);
-    // Calculate total fitting chisq.  doReserved same meaning as above.
-    double chisqDOF(int& dof, double& maxDeviate, bool doReserved=false) const;
-    void setParams(const DVector& p);
-    DVector getParams() const {return pmc.getParams();}
-    int nParams() const {return pmc.nParams();}
-    // This is the calculation of normal equation components used in fitting.
-    // reuseAlpha=true will leave alpha unchanged, way faster.
-    void operator()(const DVector& params, double& chisq,
-		    DVector& beta, DMatrix& alpha,
-		    bool reuseAlpha=false);
-    void setRelTolerance(double tol) {relativeTolerance=tol;}
-    // Return count of useful (un-clipped) Matches & Detections.
-    // Count either reserved or non-reserved objects, and require minMatches useful
-    // Detections for a valid match:
-    void count(long int& mcount, long int& dcount, bool doReserved=false, int minMatches=2) const;
-    void count(long int& mcount, long int& dcount, bool doReserved, int minMatches, long catalog) const;
-  };
-
-  // Function that will execute a Marquardt-Levenberg fit to optimize the
-  // parameters of some model to a simple list of detections.
-  // Assigns equal uncertainty to all test points, given by sigma
-  void mapFit(list<Detection*> testPoints, PixelMap* pm, double sigma=1.);
+  void clip()
+  {
+    isClipped = true;
+    fitWeight = 0.;
+  }
+};
+
+class PMDetection : public Detection
+{
+  // Derive a class which is for Gaia or other measurements that come with full 5d
+  // constraints.
+  // When filling in this object, one should also fill in the base class Detection
+  // fields as if this were a single detection at the field epoch, then this
+  // class can be sliced to a Detection and used as such.
+public:
+  EIGEN_NEW
+  PMSolution pmMean;
+  PMCovariance pmInvCov;
+
+  // Calculate full *5d* chisq relative to best prediction of itsMatch,
+  // using full fitting covariance but *before* any weighting factor is applied.
+  virtual double trueChisq() const;
```

Comment:
Overrides of methods that were defined as `virtual` in the base class should generally drop the `virtual` and add `override` to the end:
```suggestion
  double trueChisq() const override;
```